openmdao/core/implicitcomponent.py

```diff
@@ -71,6 +71,13 @@ def _solve_nonlinear(self):
             else:
                 return result
 
+    def _guess_nonlinear(self):
+        """
+        Provide initial guess for states.
+        """
+        with self._unscaled_context(outputs=[self._outputs], residuals=[self._residuals]):
+            self.guess_nonlinear(self._inputs, self._outputs, self._residuals)
```

Comment:
shouldn't we do the check here (maybe using the flag I mentioned earlier) in order to avoid unnecessary scaling operations when the system doesn't provide a guess?